src/instructlab/provenance/sigstore.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+
+# Third-party
+from sigstore.sign import SigningContext
+from sigstore.oidc import Issuer
+from sigstore.models import Bundle
+from sigstore.verify import Verifier
+from sigstore.verify.policy import Identity
+
+def sign(model_path, bundle_path, staging=False):
+    """Signs a model with Sigstore"""
+    # pylint: disable=C0415
+
+    issuer = Issuer.production()
+    if staging:
+        issuer = Issuer.staging()
+    identity = issuer.identity_token()
+
+    # The artifact to sign
+    artifact = Path(model_path).read_bytes()
```

Comment:
This approach consumes a lot of memory. Models are very large, several GB to tens of GB. I suggest that you use incremental hashing and `sigstore.hashes.Hashed` here.

Reply:
Good idea, I added an implementation I based off `sigstore-python`'s implementation of incremental hashing. Let me know how it looks.